### PR TITLE
tracing overlay

### PR DESCRIPTION
**Pull request by PietroPasotti**
added tracing overlay file

- [ ] do we need to add anything to the readme/documentation?
- [x] test the overlay

## Testing instructions
Deploy the testing and tracing overlay:

```
tox -e render-edge
juju deploy ./bundle.yaml --trust \
  --overlay overlays/tracing-overlay.yaml \
  --overlay overlays/testing-overlay.yaml
```

Then, sync latest v2 tracing lib
```
jhack sync "*" --source-dirs ./lib/charms/tempo_k8s/v2/
```
Check grafana dashboard that tempo source has charm traces from (alertmanager, prometheus, loki, grafana, tempo, traefik, catalogue)

Add tls overlay:
```
juju deploy ./bundle.yaml --trust \
  --overlay overlays/tls-overlay.yaml \
  --overlay overlays/testing-overlay.yaml
```

verify you can still see the new charm traces for all apps (and grafana workload)

### CODE REVIEW

File: overlays/tracing-overlay.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+applications:
+  tempo:
+    charm: 'tempo-k8s'
+    scale: 1
+    trust: true
+    channel: 'edge'
+
+relations:
+- [traefik:ingress, tempo:ingress]
+- [prometheus:alertmanager, alertmanager:alerting]
```
Comment (sed-i) [inline overlays/tracing-overlay.yaml:14]: Leftover?

File: overlays/tracing-overlay.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# This overlay enhances cos-lite with the distributed tracing backend Tempo.
+
+applications:
+  tempo:
+    charm: 'tempo-k8s'
+    scale: 1
+    trust: true
+    channel: 'edge'
+
+relations:
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:12]: so far none of the cos lite charms supports tracing, so I didn't think to include it yet, or will it fail nicely if it doesn't find a relation?

[the remark this answers is not in the document]
Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:12]: checked: it does not fail nicely. Added tracing relations for the charms that already support them, left the other ones out. we'll need to keep the overlay updated when we implement tracing in those

File: overlays/tracing-overlay.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- [grafana:grafana-dashboard, tempo:grafana-dashboard]
+- [grafana:grafana-source, tempo:grafana-source]
+- [prometheus:metrics-endpoint, tempo:metrics-endpoint]
+# tracing relations (monitoring cos-lite)
+- [traefik:tracing, tempo:tracing]
+- [prometheus:tracing, tempo:tracing]
+- [grafana:tracing, tempo:tracing]
+- [loki:tracing, tempo:tracing]
+- [alertmanager:tracing, tempo:tracing]
+- [catalogue:tracing, tempo:tracing]
```
Comment (sed-i) [inline overlays/tracing-overlay.yaml:25]: TLS complicates things: would need a separate overlay probably? No need at this point?
Same for the offers and 3-node overlays.

Should we have tempo overlays for cos under the tempo repo?

Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:25]: What would putting the tempo overlays in the tempo repo solve? we still have a combinatorics issue for the tls, offers, and 3-node overlays. Is it a matter of tidiness, then we could categorize them. 

```
/overlays
  /3-node
    /cos-lite
    /tracing
  /tls
    /cos-lite
    /tracing
  /offers
    /cos-lite
    /tracing
```

Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:25]: but I think bundle and overlays should live together

Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:25]: btw there's no 3-node overlay yet as tempo as it stands can't be scaled. We're working on the scaling-monolithic mode.

Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:25]: ended up putting all under /overlays/tracing

Comment (PietroPasotti) [reply overlays/tracing-overlay.yaml:25]: ended up putting all under /overlays/tracing

**Comment by mmkay** [comment]
As we now have Terraform code at https://github.com/canonical/observability/pull/209, this is not the way we'll go.

**Pull request closed without merging**